### ISSUE TITLE
Model_main on custom Faster R-CNN config and dataset fails to detect any objects

### ISSUE THREAD

**Issue by ajuschka** [tensorflow/models#6748]
Hi there, 

I already posted this on StackOverflow 
https://stackoverflow.com/questions/55991435/how-to-get-my-faster-r-cnn-model-with-a-custom-dataset-to-actually-detect-object
but could not resolve the problem.

------------------------

### System information
- **What is the top-level directory of the model you are using**: tensorflow/models/research
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04
- **TensorFlow installed from (source or binary)**: conda install  tensorflow=1.12
- **TensorFlow version (use command below)**: 1.12.0
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: -
- **GPU model and memory**: -
- **Exact command to reproduce**: python object_detection/model_main.py \
    --model_dir=PATH_TO/LogosInTheWild-v2/models/model/ \
    --pipeline_config_path=PATH_TO/LogosInTheWild-v2/models/model/faster_rcnn_inception_logos-locally-on-ubuntu.config \
    --num_train_steps=50000 --alsologtostderr

### Describe the problem
I created my own dataset with 821 classes and 10 training TFRecord files and 10 validation TFRecord files. When running model_main-py, I see in tensorboard that the ground truth is shown correctly in the images on the right but there are no detection boxes on the left. In tensorboard's scalar section loss is going up and down but maP is constantly zero.
I tried different things in my pipeline config. (Trying legacy/train.py and legacy/eval.py lead to the same results)

### Source code / logs
model {
  faster_rcnn {
    num_classes: 821
    image_resizer {
      keep_aspect_ratio_resizer {
        min_dimension: 600
        max_dimension: 1024
      }
    }
    feature_extractor {
      type: 'faster_rcnn_inception_v2'
      first_stage_features_stride: 16
    }
    first_stage_anchor_generator {
      grid_anchor_generator {
        scales: [0.25, 0.5, 1.0, 2.0]
        aspect_ratios: [0.5, 1.0, 2.0]
        height_stride: 16
        width_stride: 16
      }
    }
    first_stage_box_predictor_conv_hyperparams {
      op: CONV
      regularizer {
        l2_regularizer {
          weight: 0.0
        }
      }
      initializer {
        truncated_normal_initializer {
          stddev: 0.01
        }
      }
    }
    first_stage_nms_score_threshold: 0.0
    first_stage_nms_iou_threshold: 0.7
    first_stage_max_proposals: 300
    first_stage_localization_loss_weight: 2.0
    first_stage_objectness_loss_weight: 1.0
    initial_crop_size: 14
    maxpool_kernel_size: 2
    maxpool_stride: 2
    second_stage_box_predictor {
      mask_rcnn_box_predictor {
        use_dropout: false
        dropout_keep_probability: 1.0
        fc_hyperparams {
          op: FC
          regularizer {
            l2_regularizer {
              weight: 0.0
            }
          }
          initializer {
            variance_scaling_initializer {
              factor: 1.0
              uniform: true
              mode: FAN_AVG
            }
          }
        }
      }
    }
    second_stage_post_processing {
      batch_non_max_suppression {
        score_threshold: 0.0
        iou_threshold: 0.6
        max_detections_per_class: 200
        max_total_detections: 300
      }
      score_converter: SOFTMAX
    }
    second_stage_localization_loss_weight: 2.0
    second_stage_classification_loss_weight: 1.0
  }
}
train_config: {
  batch_size: 1
  optimizer {
    momentum_optimizer: {
      learning_rate: {
        manual_step_learning_rate {
          initial_learning_rate: 0.0002
          schedule {
            step: 900000
            learning_rate: .00002
          }
          schedule {
            step: 1200000
            learning_rate: .000002
          }
        }
      }
      momentum_optimizer_value: 0.9
    }
    use_moving_average: false
  }
  gradient_clipping_by_norm: 10.0
  #fine_tune_checkpoint: "/PATH_TO/models/model/model.ckpt"
  #from_detection_checkpoint: true
  #load_all_detection_checkpoint_vars: true
  num_steps: 50000
  
  data_augmentation_options {
    random_horizontal_flip {
    }
  }
  data_augmentation_options {
    random_vertical_flip {
    }
  }
  data_augmentation_options {
    random_rotation90 {
    }
  }
}

train_input_reader {
  label_map_path: "/PATH_TO/data/pascal_label_map.pbtxt"
  tf_record_input_reader {
    input_path:"/PATH_TO/data/pascal_train.record-?????-of-00010"
  }
}
eval_config {
  num_examples: 1508
  #max_evals: 1508
  use_moving_averages: false
  metrics_set: "pascal_voc_detection_metrics"
}
eval_input_reader {
  label_map_path:  "/PATH_TO/data/pascal_label_map.pbtxt"
  shuffle: true
  num_readers: 10
  tf_record_input_reader {
    input_path: "/PATH_TO/data/pascal_val.record-?????-of-00010"
  }
}

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be an old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
